compliance/google/long_stopped_instances/google_long_stopped_instances.pt

```diff
@@ -0,0 +1,286 @@
+# Scripts
+###############################################################################
+script "js_stoped_instances_req_list", type: "javascript" do
+ parameters "ds_stoped_instances", "param_stoped_days"  
+ result "request"
+  code <<-EOF
+  var request=[];
+ _.each(ds_stoped_instances, function(row){
+   if(row["instance_name"] != null){
+    var result = {
+        "start_at": new Date(new Date().setDate(new Date().getDate() - parseInt(param_stoped_days))).toISOString(),
+        "end_at": new Date().toISOString(),
+        "instance_name": row["instance_name"]
+      }
+    request.push(result); 
+    }
+   })
+EOF
+end
+
+script "js_report", type: "javascript" do
+  parameters "ds_stoped_instances", "ds_list_stoped_instances", "param_exclude_tags"
+  result "result"
+  code <<-EOF   
+   var result=[];
+    var param_exclude_tags_lower=[];
+    for(var i=0; i < param_exclude_tags.length; i++){
+      param_exclude_tags_lower[i]=param_exclude_tags[i].toString().toLowerCase();
+    } 
+  _.each(ds_list_stoped_instances, function(instance_uptime){
+      _.each(ds_stoped_instances, function(instance_stoped){	  
+	  var tags = instance_stoped['tags'];
+      var isTagMatched=false
+      var tagKeyValue=""
+      if(typeof(tags) !== "undefined"){
+        var tagKeys = Object.keys(tags);
+        for (var j = 0; j < tagKeys.length; j++){
+          var tagKey = tagKeys[j];
+          var tagValue = tags[tagKey];
+          //Check, if the tag present in entered param_exclude_tags, ignore the google bucket object if the tag matches/present.
+          if((param_exclude_tags_lower.indexOf(tagKey.toLowerCase()) !== -1) || (param_exclude_tags_lower.indexOf((tagKey+'='+tagValue).toLowerCase()) !== -1)){
+            isTagMatched = true;
+          }
+          if(tagValue.length > 0){
+            tagKeyValue = tagKeyValue + ', '+ tagKey+'='+tagValue
+          }else{
+            tagKeyValue = tagKeyValue + ', '+ tagKey
+          }
+        }
+      }
+	    if( instance_uptime["instance_name"] == instance_stoped["instance_name"] && instance_uptime["instance_id"] == null && !isTagMatched ){
+		  result.push({
+          "name": instance_stoped["instance_name"],
+          "status": instance_stoped["status"],
+          "instance_id": instance_stoped["instance_id"],
+          "zone": instance_stoped["zone"],
+		  "tagKeyValue":(tagKeyValue.slice(2))
+        }) 
+		} 
+	  })  
+  })
+EOF
+end
+
+###############################################################################
+# Policy
+###############################################################################
+
+policy "pol_utilization" do
+  validate_each $ds_report do
+    summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{ len data }} google cloud instances in Stopped State for Over {{parameters.param_stoped_days}} Day(s)"
+    detail_template <<-EOS
+# Google Instance Out of Compliance
+|Instance ID| Name | Status | Zone | Tags |
```

Comment:
Can we list days stopped?

Reply:
Curt, We are deciding the instance is stopped or not based on instance uptime for given period of time. So we are not getting from which date instance is stopped, so its difficult to find number of days instance is stopped. Thanks

Reply:
ok, i understand the issue.  thanks for the details.